test/parallel_api/algorithm/alg.sorting/sort_by_key.pass.cpp

```diff
@@ -0,0 +1,107 @@
+// -*- C++ -*-
+//===-- user_sort.pass.cpp ------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+#include "support/sycl_alloc_utils.h"
+
+template <sycl::usm::alloc alloc_type>
+void
+test_with_usm(sycl::queue& q)
+{
+    constexpr int N = 32;
+
+    int h_key[N] = {};
+    int h_val[N] = {};
+    for (int i = 0; i < N; i++)
+    {
+        h_val[i] = ((N - 1 - i) / 3) * 3;
+        h_key[i] = i * 10;
+    }
+
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper_h_key(q, ::std::begin(h_key), ::std::end(h_key));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper_h_val(q, ::std::begin(h_val), ::std::end(h_val));
+
+    int* d_key = dt_helper_h_key.get_data();
+    int* d_val = dt_helper_h_val.get_data();
+
+    auto first = oneapi::dpl::make_zip_iterator(d_key, d_val);
+    auto last = first + std::distance(d_key, d_key + N);
```

Comment:
do we really need `std::distance` for pointers? can we just say `first + N`?

Reply:
Fixed